lib/Dashboard/MailWidget.php

```diff
@@ -95,4 +110,72 @@ public function load(): void {
 			$this->accountService->findByUserId($this->userId)
 		);
 	}
+
+	/**
+	 * Get widget-specific search filter
+	 * @return string
+	 */
+	abstract public function getSearchFilter(): string;
+
+	/**
+	 * @param string $userId
+	 * @param int|null $minTimestamp
+	 * @param int $limit
+	 * @return Message[]
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	protected function getEmails(string $userId, ?int $minTimestamp, int $limit = 7): array {
+		$user = $this->userManager->get($userId);
+		if ($user === null) {
+			return [];
+		}
+		$filter = $this->getSearchFilter();
+		$emails = $this->mailSearch->findMessagesGlobally($user, $filter, null, $limit);
+
+		if ($minTimestamp !== null) {
+			return array_filter($emails, function (Message $email) use ($minTimestamp) {
+				return $email->getSentAt() > $minTimestamp;
+			});
+		}
```

Comment:
```suggestion
		$emails = $this->mailSearch->findMessagesGlobally($user, $filter, $minTimestamp, $limit);
```

https://github.com/nextcloud/mail/blob/a40d8751d75420635a453150b100646b623391a3/lib/Db/MessageMapper.php#L764-L768

Assuming this is the comparison we want

Reply:
Thanks but that selects all the messages that are older than a given timestamp. We want those that are more recent.